mod/log/pkg/noop/noop.go

```diff
@@ -53,3 +53,45 @@ func (n *Logger[KeyValT]) Error(string, ...KeyValT) {
 func (n *Logger[KeyValT]) Debug(string, ...KeyValT) {
 	// No operation
 }
+
+// AdvancedLogger is a Logger that performs no operations.
+type AdvancedLogger[KeyValT any, ImplT any] struct{}
```

Comment:
tbh i'd delete AdvancedLogger and just convert the other no-op to implement advanced